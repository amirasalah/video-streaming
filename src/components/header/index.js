import React from 'react'
import { Background, Container, Logo, ButtonLink, Link } from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom'

const Header = ({ bg = true, children, ...restProps }) => {
    return bg ? <Background {...restProps}>{children}</Background> : children
}
Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}
Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to}>
            <Logo {...restProps} />
        </ReactRouterLink>
    )
}
Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>
}
Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>
}
export default Header
