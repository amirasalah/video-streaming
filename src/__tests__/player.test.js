import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Player } from '../components'

describe('The player Component', () => {
    it('Renders the component with a video of a bunny!', () => {
        const { container, getByText, queryByTestId } = render(
            <Player>
                <Player.Button></Player.Button>
                <Player.Video src='/videos/bunny.mp4'></Player.Video>
            </Player>,
        )
        expect(queryByTestId('player')).toBeFalsy()
        fireEvent.click(getByText('Play'))
        expect(queryByTestId('player')).toBeTruthy()
        fireEvent.click(getByText('Play'))
        expect(queryByTestId('player')).toBeFalsy()
        expect(container.firstChild).toMatchSnapshot()
    })
})
