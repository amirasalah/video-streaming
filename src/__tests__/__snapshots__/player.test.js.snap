// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The player Component Renders the component with a video of a bunny! 1`] = `
<div
  class="player__Container-sc-1vwu8au-0"
>
  <button
    class="player__Button-sc-1vwu8au-4 leQJTN"
  >
    Play
  </button>
</div>
`;
